R/msf_dict_survey.R

```diff
@@ -21,76 +20,84 @@ msf_dict_survey <- function(disease, name = "MSF-survey-dict.xlsx",
   # read in data set - pasting the disease name for sheet
   dat_dict <- readxl::read_xlsx(path, sheet = disease)
 
+  # read in categorical variable content options
+  dat_opts <- readxl::read_xlsx(path, sheet = sprintf("%s_choices", disease))
+
   # clean col names
   colnames(dat_dict) <- tidy_labels(colnames(dat_dict))
+  colnames(dat_opts) <- tidy_labels(colnames(dat_opts))
 
-  # fill NA values with previous non-NA value, replace "." in codes and names
-  dat_dict <- tidyr::fill(dat_dict, colnames(dat_dict), .direction = "down")
-  dat_dict <- dplyr::rename_at(dat_dict,
-    .vars = dplyr::vars(dplyr::starts_with("choice_")),
-    .funs = ~ gsub("choice", "option", .)
-  )
+  # clean future var names
+  dat_dict$name <- tidy_labels(dat_dict$name)
+
+  # pre-pend "choices" to options dataset (otherwise duplicated column names on merge)
+  colnames(dat_opts) <- sprintf("choices_%s", colnames(dat_opts))
+
+  # drop rows with grouping variables for repeats (not necessary for generating data)
+  dat_dict <- dat_dict[!dat_dict$type %in% c("begin_group", "end_group", "end_repeat"), ]
 
-  # minor tidying, e.g.: create "CodeX" assignments
-  dat_dict$option_code[dat_dict$option_code == "."] <- NA
-  dat_dict$option_name[dat_dict$option_name == "."] <- NA
+
+  # minor tidying, make the type column useful as as unique ID for options
   dat_dict$type <- gsub(
-    pattern = "Question",
+    pattern = "select_one |select_multiple ",
     replacement = "",
     x = dat_dict$type
   )
 
-  # transform dat_dict to wide format (like outbreak dictionary)
-  dat_dict <- dplyr::select(
-    dat_dict,
-    !!quote(level),
-    !!quote(column_name),
-    !!quote(description),
-    !!quote(type),
-    dplyr::starts_with("option_")
-  )
+  # create counts for variable order (not entirely sure this is needed?)
+  # dat_dict <- dplyr::mutate(dat_dict, option_order_in_set = seq(dplyr::n()))
 
-  dat_dict <- dplyr::group_by(dat_dict, !!quote(column_name))
+  ## recode type to fit to DHIS2 (I dont think this is necessary?)
```

Comment:
dont think this is necessary - will circle back and delete once get through all of dict adding. Just a note to self incase forget